.github/workflows/deploy-to-production.yml

```diff
@@ -0,0 +1,125 @@
+          echo "::set-output name=id::${id}"
+        env:
+          TARGET: "prod"
+
+      - name: Get Lastest AWS Keys for Envrinment
+        id: aws-keys
+        run: |
+          id=$(echo '${{steps.files.outputs.details}}' | sed -e 's/+00:00/Z/g' | jq ' . | map( select( .name | startswith( "${{ env.TARGET }}-aws-credentials-" ) ) ) | sort_by( .createdAt | fromdate ) | reverse | .[0].id')
+          echo "::set-output name=id::${id}"
+        env:
+          TARGET: "prod"
+
+      - name: Get Lastest IP Configuration File ID
+        id: latest-ip-whitelist
+        run: |
+          id=$(echo '${{steps.files.outputs.details}}' | sed -e 's/+00:00/Z/g' | jq ' . | map( select( .name | startswith( "hippo-ipfilter-" ) ) ) | sort_by( .createdAt | fromdate ) | reverse | .[0].id')
+          echo "::set-output name=id::${id}"
+
+      - name: Get All Distribution Ids
+        id: all-distribution-id
+        run: |
+          response=$(curl \
+          -H "Authorization: Bearer ${{ steps.jwt_token.outputs.token }}" \
+          -X GET https://api.${{ secrets.HOST }}/v3/distributions)
+          echo "::set-output name=allDistIds::${response}"
+          echo ${response}
+
+      - name: Distribution Ids Retrival failed
+        if: ${{ contains(steps.all-distribution-id.response ,'401') }}
+        uses: actions/github-script@v3
+        with:
+          script: |
+            core.setFailed('Retrival of distibution ids failed ')
+
+      - name: Get Distribution Ids for prod deployment
+        id: prod-distribution-ids
+        run: |
+          latestId=$(echo '${{steps.all-distribution-id.outputs.allDistIds}}' | sed -e 's/+00:00/Z/g' | jq --compact-output '[.items[] | {id: .id , date: .createdAt, name: .name}] | map(select(.name | test("v3.0.1039.tar.gz"))) | map(.id)')
+          echo "::set-output name=prodDistId::${latestId}"
+          echo ${latestId}
+
+      - name: Request PROD Deployment from BR Cloud
+        id: deployment
+        run: |
+          code=$(curl -i \
+          -H "Accept: application/json; charset=utf-8" \
+          -H "Authorization: Bearer ${{ steps.jwt_token.outputs.token }}" \
+          -X PUT https://api.${{ secrets.HOST }}/v3/environments/${{ env.ENVIRONMENT }}/deploy \
+          -o /dev/null -w '%{http_code}\n' -s \
+          -d '{ "distributionId": "${{ env.DISTRIBUTION }}", "strategy" : "rollingupdate", "appConfigFileRoles": [ { "appConfigFileId": ${{ steps.aws-keys.outputs.id }}, "role": "file", "newFilename": "aws-credentials.properties" }, { "appConfigFileId": ${{ steps.system-properties.outputs.id }}, "role": "systemproperty" }, { "appConfigFileId": ${{ steps.latest-ip-whitelist.outputs.id }}, "role": "file", "newFilename": "hippo-ipfilter.properties" } ] }')
+          echo "::set-output name=code::$code"
```

Comment:
Are you trying to check the response header? If so, I am not convinced this check does what you expect. You can get an example of getting the response header from Verify JWT Token above. From the example, you will notice there is a subtle line missing ` -o /dev/null -w '%{http_code}\n' -s)`.